src/main/java/randoop/main/Minimize.java

```diff
@@ -0,0 +1,1185 @@
+   * @param expectedOutput expected output of running JUnit test suite
+   * @param timeoutLimit number of seconds allowed for the whole test suite to run
+   * @return true if there are no compilation and no run-time errors and the output is equal to the
+   *     expected output
+   */
+  private static boolean checkCorrectlyMinimized(
+      String filePath,
+      String classpath,
+      String packageName,
+      Map<String, String> expectedOutput,
+      int timeoutLimit) {
+    // Check that the exit value from compiling the Java file is zero.
+    if (compileJavaFile(filePath, classpath, packageName, timeoutLimit) != 0) {
+      return false;
+    }
+
+    // Run the Java file and get the standard output.
+    String runResult = runJavaFile(filePath, classpath, packageName, timeoutLimit);
+
+    // Compare the standard output with the expected output.
+    return expectedOutput.equals(normalizeJUnitOutput(runResult));
+  }
+
+  /**
+   * Compile a Java file and return the compilation exit value.
+   *
+   * @param filePath the absolute path to the Java file to be compiled and executed
+   * @param classpath dependencies and complete classpath to compile and run the Java program
+   * @param packageName the package that the Java file is in
+   * @param timeoutLimit number of seconds allowed for the whole test suite to run
+   * @return exit value of compiling the Java file
+   */
+  private static int compileJavaFile(
+      String filePath, String classpath, String packageName, int timeoutLimit) {
+    // Obtain directory to carry out compilation and execution step.
+    String executionDir = getExecutionDirectory(filePath, packageName);
+
+    // Command to compile the input Java file
+    String command = "javac -classpath " + systemClassPath;
+    // Add current directory to class path
+    command += pathSeparator + ".";
+    if (classpath != null) {
+      // Add specified classpath to command
+      command += pathSeparator + classpath;
+    }
+    command += " " + filePath;
+
+    // Compile specified Java file.
+    return runProcess(command, executionDir, timeoutLimit).exitValue;
+  }
+
+  /**
+   * Run a Java file and return the standard output.
+   *
+   * @param filePath the absolute path to the Java file to be compiled and executed
+   * @param classpath dependencies and complete classpath to compile and run the Java program
+   * @param packageName the package that the Java file is in
+   * @param timeoutLimit number of seconds allowed for the whole test suite to run
+   * @return the standard output from running the Java file
+   */
+  private static String runJavaFile(
+      String filePath, String classpath, String packageName, int timeoutLimit) {
+    // Obtain directory to carry out compilation and execution step.
+    String executionDir = getExecutionDirectory(filePath, packageName);
+
+    // Obtain directory path from file path.
+    Path fPath = Paths.get(filePath).getParent();
```

Comment:
What does `fPath` stand for?  I would have thought it had to do with the file, but it's actually the file's directory.

It's confusing that variables named `*Path` are sometimes strings and sometimes `Path`s.  Could you clean this up throughout the code?  It would be nice to do less string manipulation throughout the code.  For example, `writeToFile` could return a `File` rather than a `String`.